### PR TITLE
Extract stdlib/ to luna/luna-stdlib

### PR DESCRIPTION
**Pull request by Spaceghost**
# Outstanding tasks

- [x] Extract `stdlib/` to `luna-develop/apps/stdlib`
- [ ] Preserve Haskell src from `stdlib/`
- [ ] Fix hard-coded paths referencing repo parent-directory
- [ ] Update README.md and any extraneous documentation
- [ ] Update tooling to expect `stdlib` in `luna-develop/apps/`
- [ ] Create `luna/luna-stdlib`
- [ ] Follow README.md in `Spaceghost/luna-stdlib` instead of forking
- [ ] Merge this PR!

This will address #50 

Signed-off-by: Johnneylee Jack Rollins <Johnneylee.Rollins@gmail.com>

### CODE REVIEW

File: passes/stack.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ ghc-options: {'*': -fenable-rewrite-rules -threaded -funbox-strict-fields -fcons
 packages:
 - {location: .}
 - {location: ../core}
-- {location: ../stdlib}
+- {location: ../../stdlib}
```
Comment (Spaceghost) [inline passes/stack.yaml]: This is quite gross to me, but I'm not sure how to solve it. It serves the luna-manager develop directory layout, but is an eyesore. Can I get a review?

**Comment by wdanilo** [comment]
If we want to keep Luna STD lib separately from the compiler (and that makes sense to me), we should use here `LUNAPATH` env var to show Luna all the paths it should look up for libraries, including std library. 

However, looking at the changes, I see that you've also removed `Luna/Prim/*` files and Luna bindings defined in Haskell. `Luna/Prim/*` should never be removed, these are primitive types, like `Int` and they need to be implemented in the compiler, not library. All the .hs bindings have to be here too, unless we provide proper FFI. Please consult such changes with @kustosz before making them - after checking the changes I'm sure you two were not talking with each other, right?

Basically the structure is very gentle thing and we should discuss very much how it would look like before making such changes. As I described somewhere else, this means in particular that relative paths cannot be hard-coded and point over root library folder, like now. Even if such commit appears, we should have a clear path how to make them correct (e.g. by using env variables), which is impossible when moving Luna/Prim or .hs bindings outside of the compiler.

**Comment by kustosz** [comment]
@wdanilo said all there's to be said – the idea is to only have the `.luna` files split out into a separate repo, the lowest level haskell bindings should remain a part of `luna` repository. At some point, this new repository will have a standard Luna library form, which won't usually contain such low level bindings. Only such bindings will happen via the FFI inside libs.

**Comment by Spaceghost** [comment]
I have added a task to preserve the Haskell-y files in the `luna/luna` tree.
I will separate the two and keep the Luna source in the new proposed `luna/luna-stdlib` repository.

As for the `$LUNAPATH` I'd like someone else to ultimately handle writing this code, but I believe it would not prevent extraction of the stdlib. The build system picks up the stdlib in `luna-develop/apps/stdlib` for me for now.

**Comment by wdanilo** [comment]
@Spaceghost before you move with this topic further, please design it really well with Marcin. Task for preserving the files is not good enough, because neither of us (me, you, Marcin) see the big picture. First of all you guys have to discuss it in detail and then do it step by step. By the way I dont know if this is really the highest priority for now, because it will be MUCH easier when pkgmanager will be here.
